src/xr/xr-depth-sensing.js

```diff
@@ -0,0 +1,281 @@
+import { EventHandler } from '../core/event-handler.js';
+import { Mat4 } from '../math/mat4.js';
+import { Texture } from '../graphics/texture.js';
+import { ADDRESS_CLAMP_TO_EDGE, PIXELFORMAT_L8_A8 } from '../graphics/graphics.js';
+
+/**
+ * @class
+ * @name pc.XrDepthSensing
+ * @augments pc.EventHandler
+ * @classdesc Depth Sensing provides depth information which is reconstructed using underlying AR system. It provides ability to query depth value (CPU path) or access a depth texture (GPU path). Depth information can be used (not limited to) for: reconstructing real world geometry; virtual object placement; occlusion of virtual objects by real world geometry; and more.
+ * @description Depth Sensing provides depth information which is reconstructed using underlying AR system. It provides ability to query depth value (CPU path) or access a depth texture (GPU path). Depth information can be used (not limited to) for: reconstructing real world geometry; virtual object placement; occlusion of virtual objects by real world geometry; and more.
+ * @param {pc.XrManager} manager - WebXR Manager.
+ * @property {boolean} supported True if Depth Sensing is supported.
+ * @property {number} width Width of depth texture or 0 if not available.
+ * @property {number} height Height of depth texture or 0 if not available.
+ * @example
+ * // CPU path
+ * var depthSensing = app.xr.depthSensing;
+ * if (depthSensing.available) {
+ *     // get depth in the middle of the screen, value is in meters
+ *     var depth = depthSensing.getDepth(depthSensing.width / 2, depthSensing.height / 2);
+ * }
+ * @example
+ * // GPU path, attaching texture to material
+ * material.diffuseMap = depthSensing.texture;
+ * material.setParameter('matrix_depth_uv', depthSensing.uvMatrix.data);
+ * material.update();
+ *
+ * // update UV transformation matrix on depth texture resize
+ * depthSensing.on('resize', function () {
+ *     material.setParameter('matrix_depth_uv', depthSensing.uvMatrix.data);
+ * });
+ * @example
+ * // // GLSL shader to unpack depth texture
+ * // varying vec2 vUv0;
+ * //
+ * // uniform sampler2D texture_depthSensingMap;
+ * // uniform mat4 matrix_depth_uv;
+ * //
+ * // void main(void) {
+ * //     // transform UVs using depth matrix
+ * //     vec2 texCoord = (matrix_depth_uv * vec4(vUv0.xy, 0.0, 1.0)).xy;
+ * //
+ * //     // get luminance alpha components from depth texture
+ * //     vec2 packedDepth = texture2D(texture_depthSensingMap, texCoord).ra;
+ * //
+ * //     // unpack into single value in millimeters
+ * //     float depth = dot(packedDepth, vec2(255.0, 256.0 * 255.0)); // mm
+ * //
+ * //     // normalize: 0m to 8m distance
+ * //     depth = min(depth / 8000.0, 1.0); // 0..1 = 0m..8m
+ * //
+ * //     // paint scene from black to white based on distance
+ * //     gl_FragColor = vec4(depth, depth, depth, 1.0);
+ * // }
```

Comment:
Why is all the example code commented out?

Reply:
Because it is invalid javascript (glsl here), so it will complain during generation of jsdocs.